Store RTC Engine metrics (#1)
membraneframework/membrane_rtc_engine_timescaledb: README.md

[![API Docs](https://img.shields.io/badge/api-docs-yellow.svg?style=flat)](https://hexdocs.pm/membrane_rtc_engine_timescaledb)
[![CircleCI](https://circleci.com/gh/membraneframework/membrane_rtc_engine_timescaledb.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_rtc_engine_timescaledb)

This repository contains functions, that allows storing `Membrane.RTC.Engine` metrics reports in a database.

It is part of [Membrane Multimedia Framework](https://membraneframework.org).

## Installation

The package can be installed by adding `membrane_rtc_engine_timescaledb` to your list of dependencies in `mix.exs`:

```elixir
def deps do
  [
    {:membrane_rtc_engine_timescaledb, "~> 0.1.0"}
  ]
end
```

## Usage

To use `Membrane.RTC.Engine.TimescaleDB`, put the following line in your `config` file: 
```elixir
config :membrane_rtc_engine_timescaledb, repo: MyApp.Repo, cleanup_interval: 60 * 60, metrics_lifetime: 60 * 60 * 24
```
where 
 * `repo` is a module in your project, that uses `Ecto.Repo`
 * `cleanup_interval` is the number of seconds between database cleanups 

>  * `metrics_lifetime` is the number of seconds, that says, how old records will be deleted during cleanup

Reviewer: too, many, commas
Anyway, I'd rephrase it to something like
```suggestion
`metrics_lifetime` is the number of seconds that must pass from creation before each metric can be deleted during cleanup.
```

> `:repo` option is required, but `:cleanup_interval` and `:metrics_lifetime` have default values, equal to these in the example line above

Reviewer: Instead of the last sentence, include the info in the options' descriptions, i.e.
```suggestion
where 
 * `repo` (required) is a module in your project, that uses `Ecto.Repo`
 * `cleanup_interval` (default: 1 hour) is the number of seconds between database cleanups 
 * `metrics_lifetime` (default: 24 hours) is the number of seconds, that says, how old records will be deleted during cleanup
```

[![API Docs](https://img.shields.io/badge/api-docs-yellow.svg?style=flat)](https://hexdocs.pm/membrane_rtc_engine_timescaledb)
[![CircleCI](https://circleci.com/gh/membraneframework/membrane_rtc_engine_timescaledb.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_rtc_engine_timescaledb)

This repository contains functions, that allows storing `Membrane.RTC.Engine` metrics reports in a database.

It is part of [Membrane Multimedia Framework](https://membraneframework.org).

## Installation

The package can be installed by adding `membrane_rtc_engine_timescaledb` to your list of dependencies in `mix.exs`:

```elixir
def deps do
  [
    {:membrane_rtc_engine_timescaledb, "~> 0.1.0"}
  ]
end
```

## Usage

To use `Membrane.RTC.Engine.TimescaleDB`, put the following line in your `config` file: 
```elixir
config :membrane_rtc_engine_timescaledb, repo: MyApp.Repo, cleanup_interval: 60 * 60, metrics_lifetime: 60 * 60 * 24
```
where 
 * `repo` (required) is a module in your project, that uses `Ecto.Repo`
 * `cleanup_interval` (default: 1 hour) is the number of seconds between database cleanups 

>  * `metrics_lifetime` (default: 24 hours) is the number of seconds that must pass from creation before each metric can be deleted during cleanup

Reviewer: This is incomplete, to use it the user must apply the migrations. You should mention that it requires postgresql with timescaledb extension, a configured Ecto repo and then the migration in their project calling the ones from library

Reviewer: In other words: ensure anyone using rtc engine can set up this metrics following the instructions found in readme
